http-clients/src/main/java/org/triplea/http/client/web/socket/WebSocketConnection.java

```diff
@@ -89,16 +86,14 @@ void close() {
   void connect() {
     Preconditions.checkState(!client.isOpen());
     Preconditions.checkState(!closed);
-    webSocketConnector.initiateConnection();
+    Interruptibles.await(
+        () -> client.connectBlocking(DEFAULT_CONNECT_TIMEOUT_MILLIS, TimeUnit.MILLISECONDS));
```

Comment:
I hardcoded the timeout here, if you think making this configurable for testing is better, let me know

Reply:
Also `connectBlocking()` would return false if anything goes wrong according to the documentation, I wonder if we should use that to allow for better error handling

Reply:
It might make sense to let the integration test land before we update this too much. With that test we'll be able to be sure we have feature parity.

Correct me if I'm mistaken here, I think we may run into an issue with 'send'. The `GenericWebSocketClient` sees the connect as blocking, but it's free to call 'send' whenever it wants. If we then get into the send code, the connection might not yet be open.

Reply:
Let me clear things up for you:
`WebSocketConnection#connect` is blocking so in a single threaded environment if that method is called on initialisation, it blocks until the connection is either established or failed, so any subsequent calls to send occur after a successful or unsuccessful connection, but the state is unambiguous.
So here's the thing with `GenericWebsocketClient`: In the constructor it adds the blocking connect call to the single threaded executor, in an asynchronous way, so the constructor itself isn't blocking, which is the goal.
So whenever there is a call to send before the connection has been established, this send action will be added to the executor queue as well, but because the executor is single-threaded it will not be executed before the initial blocking connection action has been finished.
This obviously doesn't allow us to use more than 1 Thread in the executor, but if we ever get to the point where 1 thread is too low for the amount of incoming messages we have a completely different problem anyways.

Reply:
Okay, that makes sense to queue the requests and have connection go first. There are some updates needed:
- javadoc of the class containing that threadpool should document that sequencing
- GenericWebsocketClient should guard against 'send' being called before 'connect'. It seems like if we get a call in that ordering, that send would fail or block, when before it could have been called at the same time or out of order
- The javadoc of this class should be updated as well, it's out of date
- The TODO statement on the 'send' method in generic websocket is out of date, we now require the single threaded threadpool. Perhaps commenting on the threadpool that it's single threaded to be effectively a blocking queue. 
- I suspect the test cases should be updated to check for the new behavior. 

That is hopefully relatively exhaustive of a list. Again I'd like to see this class and set of updates be run against the in-flight integ test that verifies a chat flow.